phoenix-core/src/main/java/org/apache/phoenix/query/StatsLoaderImpl.java

```diff
@@ -47,29 +49,37 @@ public GuidePostsInfo loadStats(GuidePostsKey statsKey) throws Exception {
     public GuidePostsInfo loadStats(GuidePostsKey statsKey, GuidePostsInfo prevGuidepostInfo) throws Exception {
         assert(prevGuidepostInfo != null);
 
-        TableName tableName = SchemaUtil.getPhysicalName(
-                PhoenixDatabaseMetaData.SYSTEM_STATS_NAME_BYTES,
-                queryServices.getProps());
-        Table statsHTable = queryServices.getTable(tableName.getName());
+        Optional<Connection> connection = Optional.absent();
 
         try {
-            GuidePostsInfo guidePostsInfo = StatisticsUtil.readStatistics(statsHTable, statsKey,
-                    HConstants.LATEST_TIMESTAMP);
-            traceStatsUpdate(statsKey, guidePostsInfo);
-            return guidePostsInfo;
-        } catch (TableNotFoundException e) {
-            // On a fresh install, stats might not yet be created, don't warn about this.
-            LOGGER.debug("Unable to locate Phoenix stats table: " + tableName.toString(), e);
-            return prevGuidepostInfo;
-        } catch (IOException e) {
-            LOGGER.warn("Unable to read from stats table: " + tableName.toString(), e);
-            return prevGuidepostInfo;
-        } finally {
+            connection = statsTableProvider.openConnection();
```

Comment:
If we use JVM cache, in my understanding, we should create connection just once and keep this connection during the whole life cycle of StatsLoaderImpl object. With your implementation, we use TransientConnectionStatsTableProvider, so we create connection and close the connection for each time that loadStats is called -- I'm wondering whether this is too expensive.